### ISSUE TITLE
[test] Add an e2e or integration test to cover the issue 

### ISSUE THREAD

**Issue by ahrtr** [etcd-io/etcd#13715]
The issue [13466](https://github.com/etcd-io/etcd/issues/13466) is  resolved by PR [13645](https://github.com/etcd-io/etcd/pull/13645), but there is no e2e or integration test yet, so raising this ticket to track it.

**Comment by serathius** [comment]
We should also add unit test.

**Comment by moficodes** [comment]
I would like to take a try at it!

**Comment by jmhbnz** [comment]
Awesome thanks @moficodes - Assigned your way!

/assign @moficodes

**Comment by amosehiguese** [comment]
@ahrtr I believe this is still available to be worked on? 

I will love to take this up. If there are any other useful information, please kindly let me know.

**Comment by ahrtr** [comment]
> [@ahrtr](https://github.com/ahrtr) I believe this is still available to be worked on?

Yes, refer to https://github.com/ahrtr/etcd-issues/tree/master/issues/13466

Thanks

/assign @amosehiguese

**Comment by amosehiguese** [comment]
Also, since there will be units, integration, and e2e tests, should they all come as separate PRs or as a single PR? 

...

> Yes, refer to https://github.com/ahrtr/etcd-issues/tree/master/issues/13466
>
Will do. Thanks 

@ahrtr
